### PR TITLE
Fix for WFCORE-4923, resolve expression resursively which causes improper behaviour of the jboss CLI for jberet batch

### PR DESCRIPTION
**Pull request by swetapatra**
Fix for:
https://issues.redhat.com/browse/WFCORE-4923
resolve expression resursively which causes improper behaviour of the jboss CLI

### CODE REVIEW

**Comment by swetapatra** [comment]
> When we submit we need to make sure two different areas are covered: -
> 
>     1. The code needs to be re-viewable to reviewers can see what is being fixed / implemented.
> 
>     2. In the future the git history should identify why changes were made.
> 
> 
> Looking at old history full of messages like "Fix for WFCORE-1111", "Fix for WFCORE-2222" can be very difficult to see the history without manually opening up each Jira issues.
> 
> Can you please update the commit message so it contains the Jira issue AND a short sentence describing what is actually being fixed.
> 
> Also can you please update the PR with the same information AND include the link to the Jira issue.

I am done with the requested changes will keep these things pointed out to me in mind while raising pull requests in future

**Comment by swetapatra** [comment]
> Code looks fine, thanks.
> 
> However, @darranl was suggesting to update the _commit message_, not the GitHub PR as it is the commits that are read when looking at the history.
> 
> Could you please amend your commit message with the appropriate message and a link to https://issues.redhat.com/browse/WFCORE-4923? thanks!

Hi,I have edited the commit message to reflect the issue and also linked the bug to it

**Comment by swetapatra** [comment]
Do I have to build this again?Or will you guys take a look?

**Comment by darranl** [comment]
The commit message has not been updated as previously requested, once that is updated it will automatically trigger another round of CI.

**Comment by darranl** [comment]
That git update looks like it went wrong, somehow you seem to have rebased a load of commits and made them look new - from the original branch the command "git commit --amend" should be sufficient to edit the commit message.

**Comment by swetapatra** [comment]
> That git update looks like it went wrong, somehow you seem to have rebased a load of commits and made them look new - from the original branch the command "git commit --amend" should be sufficient to edit the commit message.

That's the only command I used,but seems like there was some conflict.
Let me try again.

**Comment by swetapatra** [comment]
I am closing this pull request as somehow it involved huge upstream fetch commits and the branch is showing huge number of changes,even stash is not working,will create a new request in few minutes and send it,containing the change only
